common/selectors/config.ts

```diff
@@ -4,13 +4,69 @@ import {
   NodeConfig,
   CustomNodeConfig,
   CustomNetworkConfig,
-  Token
-} from 'config/data';
+  Token,
+  WalletName,
+  SecureWalletName,
+  InsecureWalletName,
+  NETWORKS,
+  DPathFormats
+} from 'config';
 import { INode } from 'libs/nodes/INode';
 import { AppState } from 'reducers';
 import { getUnit } from 'selectors/transaction/meta';
 import { isEtherUnit } from 'libs/units';
 import { SHAPESHIFT_TOKEN_WHITELIST } from 'api/shapeshift';
+import { DPath } from 'config/dpaths';
+import sortedUniq from 'lodash/sortedUniq';
+
+type PathType = keyof DPathFormats;
+
+type DPathFormat =
+  | SecureWalletName.TREZOR
+  | SecureWalletName.LEDGER_NANO_S
+  | InsecureWalletName.MNEMONIC_PHRASE;
+
+export function getPaths(pathType: PathType): DPath[] {
+  const paths: DPath[] = [];
+  Object.values(NETWORKS).forEach(networkConfig => {
+    const path = networkConfig.dPathFormats ? networkConfig.dPathFormats[pathType] : [];
+    if (path) {
+      paths.push(path as DPath);
+    }
+  });
+  return sortedUniq(paths);
+}
+
+export function getSingleDPath(format: DPathFormat, network: NetworkConfig): DPath {
+  const dPathFormats = network.dPathFormats;
+  return dPathFormats[format];
+}
+
+export function isSupportedWalletFormat(format: WalletName, network: NetworkConfig): boolean {
+  const CHECK_FORMATS: DPathFormat[] = [
+    SecureWalletName.LEDGER_NANO_S,
+    SecureWalletName.TREZOR,
+    InsecureWalletName.MNEMONIC_PHRASE
+  ];
+
+  const isHDFormat = (f: string): f is DPathFormat => CHECK_FORMATS.includes(f as DPathFormat);
+
+  // Ensure DPath's are found
+  if (isHDFormat(format)) {
+    return !!(network.dPathFormats && network.dPathFormats[format]);
+  }
+
+  // Ensure Web3 is only enabled on ETH
+  // TODO -- determine if we can select testnets via MetaMask
+  if (format === SecureWalletName.WEB3) {
+    if (network.name !== 'ETH') {
+      return false;
+    }
+  }
+
+  // All other wallet formats are supported
+  return true;
+}
```

Comment:
These are all handy, but I'd call these utils, not selectors. Selectors are a redux concept that are only supposed to take in state, and return a subset of the state.

Reply:
Agreed, actually debated this with @HenryNguyen5 and we were both uncertain. I can move it to a utility file since you've brought it up as well.